subsys/bluetooth/host/conn.c

```diff
@@ -383,18 +383,12 @@ static void wait_for_tx_work(struct bt_conn *conn)
 #if defined(CONFIG_BT_CONN_TX)
 	LOG_DBG("conn %p", conn);
 
-	if (IS_ENABLED(CONFIG_BT_RECV_WORKQ_SYS)) {
```

Comment:
Thanks! 
can you use that instead? The compiler can then remove the if entirely when `WORKQ_SYS`:
```c
	if (IS_ENABLED(CONFIG_BT_RECV_WORKQ_SYS) ||
	    k_current_get() == k_work_queue_thread_get(&k_sys_work_q)) {
		tx_notify(conn);
```